src/NationalInstruments.Analyzers/Correctness/DisposableOmitsFinalizerAnalyzer.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NationalInstruments.Analyzers.Properties;
+using NationalInstruments.Analyzers.Utilities;
+using NationalInstruments.Analyzers.Utilities.Extensions;
+
+namespace NationalInstruments.Analyzers.Correctness
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DisposableOmitsFinalizerAnalyzer : NIDiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "NI1816X";
+
+        public static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            new LocalizableResourceString(nameof(Resources.NI1816X_Title), Resources.ResourceManager, typeof(Resources)),
+            new LocalizableResourceString(nameof(Resources.NI1816X_MessageFormat), Resources.ResourceManager, typeof(Resources)),
+            category: "NationalInstruments",
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            description: new LocalizableResourceString(nameof(Resources.NI1816X_Description), Resources.ResourceManager, typeof(Resources)));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterSymbolAction(AnalyzeSymbol, SymbolKind.NamedType);
+        }
+
+        private static void AnalyzeSymbol(SymbolAnalysisContext context)
+        {
+            var namedTypeSymbol = (INamedTypeSymbol)context.Symbol;
+
+            if (namedTypeSymbol.IsOrInheritsFromClass("NationalInstruments.Core.Disposable"))
```

Comment:
Looking through some of the analyzers we have in the "correctness" category, it seems there's a lot of ASW specific stuff in there - which is perhaps to be expected given the origins of this analyzer. I suppose one more wouldn't hurt... though I would really like to see us break the implicit ASW dependency here sometime.

Reply:
I agree that we should look to break the implicit dependency to ASW at some point. Up until now, we assumed this analyzer was used in ASW and I believe there might be a few other NI-specific analyzers that are making certain assumptions. We originally designed this github project to be a single package that you can import and get all of the functionality for ASW, but we might be at the point where we need to separate things out now that we have more usage outside of ASW.

One possibility is to have two different projects, an ASW-specific one that imports this package. It could then override some settings and add more analyzers. We'd need to look at the cost-benefit of doing that. @nick-beer feel free to ping me and we can talk about that at some point.

Reply:
Since this discussion is for how to split the analyzers later, I'm going to mark it resolved for now, so it doesn't gate this addition from going in.

Reply:
> I think it would make sense to have the ASW analyzer _in_ ASW, rather than github

Yeah. In this case, we'd want NationalInstruments.Analyzers.Utilities to be its own NuGet so we're not duplicating code.

Reply:
@nick-beer I went ahead and marked it disabled by default, and someone can show me how to enable it when we pull it into ASW. Also, the reasoning at the time of Jonathan's PR that you referenced (for them to be different) was based on us using .NET framework 4.6.2 at the time. Since then, NI has changed to shipping .NET 4.8 instead. So I think we might want to revisit that decision.

Reply:
@dmondrik - true, however, I do _not_ expect us to update our references to .NET 4.8 - even though we'll be installing .NET 4.6.2. Updating would be a pretty good effort, and our stated position on this has been that we'd rather spend any/all of that time migrating to modern .NET rather than making updates to our .NET Framework projects.